tests/suite/reducer/duplicate.yaml

```diff
@@ -0,0 +1,8 @@
+zql: count(), count()
```

Comment:
We could also detect this statically and make it a compile-time error (or alternatively, don't make it an error, but automatically rename the second field to something like `count_1`, though I'm not sure what the use case would be...).

Reply:
Given that we can (in theory) manually write `count2 = count()` etc, I have a slight preference for making this an error and having users manually rewrite names if that's what they want.
Re detecting statically, that does seem more user-friendly, but I think that the changes here are a good safety net -- they'll also catch file inputs that repeat column names, other procs that we haven't thought through to add static checking to, etc.  I propose merging this and adding static checking in reducer/groupby in a separate patch?

Reply:
That sounds good - I agree that we need this change anyway, since it's a bug that LookupTypeRecord doesn't check for duplicate names.